dvc/exceptions.py

```diff
@@ -338,3 +338,11 @@ def __init__(self, url, cause=None):
             ),
             cause=cause,
         )
+
+
+class HTTPErrorStatusCodeException(DvcException):
+    def __init__(self, code, reason):
+        super(HTTPErrorStatusCodeException, self).__init__(
+            "Server responded with error status code: '{}' and message: "
+            "'{}'".format(code, reason)
```

Comment:
Code and reason should go next to each other, separate by space. This how it is presented everywhere. We might also want to say that this is an HTTP error in the message.